### PR TITLE
fw-ectool: unstable-2022-12-03 -> 0-unstable-2023-12-15, switch upstream

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/fw-ectool/default.nix [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
 , pkg-config
-, hostname
+, libusb1
+, libftdi1
 }:
 
 stdenv.mkDerivation {
   pname = "fw-ectool";
-  version = "unstable-2022-12-03";
+  version = "0-unstable-2023-12-15";
```
Comment (leona-ya) [inline pkgs/os-specific/linux/fw-ectool/default.nix:12]: why do you prefix this with `0-`? it seems to work for me without it too.

**Review by leona-ya — COMMENTED** [review]
Otherwise tested and looks good.

File: pkgs/os-specific/linux/fw-ectool/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/os-specific/linux/fw-ectool/default.nix:12]: Due to the new versioning scheme documented in https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

**Comment by NickCao** [comment]
Tested most functionalities, e.g. changing fan curves, checking battery.

**Comment by io12** [comment]
This was a breaking change for me (Framework 13th gen Intel).

```
$ sudo nix run nixpkgs/nixos-23.11#fw-ectool led power red
$ sudo nix run nixpkgs#fw-ectool led power red            
Missing Chromium EC memory map.
Cannot find I2C adapter
Unable to establish host communication
Couldn't find EC
$ 
```

**Comment by NickCao** [comment]
> This was a breaking change for me (Framework 13th gen Intel).

May you try different interfaces? e.g. `ectool --interface lpc led power red`

**Comment by io12** [comment]
> > This was a breaking change for me (Framework 13th gen Intel).
> 
> May you try different interfaces? e.g. `ectool --interface lpc led power red`

Same issue (the older version with `lpc` works, but not the newer one)

```
$ sudo nix run nixpkgs#fw-ectool -- --interface dev led power off
Unable to establish host communication
Couldn't find EC
$ sudo nix run nixpkgs#fw-ectool -- --interface i2c led power off
Cannot find I2C adapter
Unable to establish host communication
Couldn't find EC
$ sudo nix run nixpkgs#fw-ectool -- --interface lpc led power off
Missing Chromium EC memory map.
Unable to establish host communication
Couldn't find EC
$ sudo nix run nixpkgs/nixos-23.11#fw-ectool -- --interface dev led power off
Unable to establish host communication
Couldn't find EC
$ sudo nix run nixpkgs/nixos-23.11#fw-ectool -- --interface i2c led power off
Cannot find I2C adapter
Unable to establish host communication
Couldn't find EC
$ sudo nix run nixpkgs/nixos-23.11#fw-ectool -- --interface lpc led power off
$ 
```

**Comment by NickCao** [comment]
Let's revert this for now.

**Comment by EricTheMagician** [comment]
> This was a breaking change for me (Framework 13th gen Intel).
> 
> ```
> $ sudo nix run nixpkgs/nixos-23.11#fw-ectool led power red
> $ sudo nix run nixpkgs#fw-ectool led power red            
> Missing Chromium EC memory map.
> Cannot find I2C adapter
> Unable to establish host communication
> Couldn't find EC
> $ 
> ```

I tried asking upstream and this was the response:
```
> Can someone test this command on a 13th gen intel cpu?
> sudo ectool led power red .
> ...

Unfortunately, the modern devices don’t expose support for red on the power LED. 
That having been said, you still shouldn’t get that error. 
Can you check whether the module cros_ec_lpcs is loaded? 
If not, no EC communication will work. :slight_smile:
```
Source: [official framework forum](https://community.frame.work/t/exploring-the-embedded-controller/12846/187)

**Comment by io12** [comment]
The `cros_ec_lpcs` module shows up in `lsmod`. I'm not sure what changed, but I tried again with the same package and same commit hash, and also by building the upstream code from source, and both times it worked. Maybe some driver update somewhere fixed it? Anyway, now I don't need to worry about upstream breaking https://github.com/io12/nixos-framework-led.

**Comment by EricTheMagician** [comment]
@NickCao 
Can we bring back this package and update it to the latest version?

**Comment by NickCao** [comment]
> @NickCao Can we bring back this package and update it to the latest version?

Revived in https://github.com/NixOS/nixpkgs/pull/312063, please test.

**Comment by io12** [comment]
> > @NickCao Can we bring back this package and update it to the latest version?
> 
> Revived in #312063, please test.

I tested and it works for everything I use it for.

**Pull request merged**